### DISCUSSION TITLE
How the ACH returns work for prenotifications?

### DISCUSSION THREAD

**Discussion by samyje** [moov-io/ach#995] [Q&A] [answered]
We are trying to integrate prenotifications in our system to be compliant with latest regulations from NACHA. It is understandable that we have to send NACHA format file with necessary transaction codes to the bank for prenotification. In return, if the added bank detail is not correct bank will send us the return code. So far so good. But how will bank send us the file? Is it in the same NACHA format? Its nearly impossible to find about the functional and technical process of NACHA prenotification. Your help on this will really be helpful.

**Comment by adamdecaf** [comment, marked as answer]
Typically they will send a "return" file on the SFTP server when errors occur (account not found). This is usually in a separate directory from where you upload files. The return file has `EntryDetail` records with an `Addenda99` record. 

Often these return files will be missing File Header/Control or Batch Header/Control records, so adjusting `ach.ValidateOpts` when reading helps to fully parse what is available.

**Reply by samyje** [reply]
Thanks a lot @adamdecaf :)
